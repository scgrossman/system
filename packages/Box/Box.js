import React from "react";
import PropTypes from "prop-types";

import styles from "./Box.scss";

const Box = ({ children, onClick }) => {
  return (
    <div onClick={onClick} className={styles.header}>
      {children}
    </div>
  );
};

Box.propTypes = {
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func
};

Box.defaultProps = {
  onClick: () => {}
};

export default Box;
