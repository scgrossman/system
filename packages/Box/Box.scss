.header {
    border: .05rem solid #eee;
    border-radius: .5rem;
    box-shadow: .0rem 1rem .9rem rgba(0,0,0,0.07);
    font-family: Arial, Helvetica, sans-serif;
    padding: 4rem;
    color: #aaa;
    &:hover {
        background-color: rgb(80, 27, 80);
    }
}