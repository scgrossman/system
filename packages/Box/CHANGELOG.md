# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.2.0](https://github.com/scgrossman/system/compare/@rdmsystem/box@1.1.0...@rdmsystem/box@1.2.0) (2019-05-23)


### Features

* **Box:** update hover color ([76aaa4e](https://github.com/scgrossman/system/commit/76aaa4e))





# 1.1.0 (2019-05-23)


### Features

* **box:** add hover ([bb447b8](https://github.com/scgrossman/system/commit/bb447b8))
