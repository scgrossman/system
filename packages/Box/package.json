{
  "name": "@rdmsystem/box",
  "version": "1.4.0",
  "description": "A test box component",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "prop-types": "^15.6.2"
  }
}
