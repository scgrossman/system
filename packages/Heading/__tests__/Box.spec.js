import React from "react";
import { shallow } from "enzyme";

import Box from "../Box";

describe("Box", () => {
  const doShallow = props => shallow(<Box {...props}>Some content</Box>);

  it("renders", () => {
    const box = doShallow();
    expect(box).toMatchSnapshot();
  });

  it("defaultProp click", () => {
    const clicked = Box.defaultProps.onClick;
    expect(clicked).toBeDefined();
  });
});
