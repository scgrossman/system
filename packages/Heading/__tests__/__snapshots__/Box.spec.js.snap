// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders 1`] = `
<div
  className="header"
  onClick={[Function]}
>
  Some content
</div>
`;
